packages/plugin-ext/src/main/browser/workspace-main.ts

```diff
@@ -227,6 +233,59 @@ export class WorkspaceMainImpl implements WorkspaceMain, Disposable {
         return uriStrs.map(uriStr => Uri.parse(uriStr));
     }
 
+    async $findTextInFiles(query: theia.TextSearchQuery, options: theia.FindTextInFilesOptions, searchRequestId: number,
+        token: theia.CancellationToken = CancellationToken.None): Promise<theia.TextSearchComplete> {
+        const defaultMaxHits = 150;
+        return new Promise(resolve => {
+            let matches = 0;
+            const what: string = query.pattern;
+            const rootUris = this.roots.map(r => r.uri);
+            this.searchInWorkspaceService.searchWithCallback(what, rootUris, {
+                onResult: (searchId, result) => {
+                    const hasSearch = this.workspaceSearch.has(searchId);
+                    if (!hasSearch) {
+                        token.onCancellationRequested(() => {
+                            this.searchInWorkspaceService.cancel(searchId);
+                        });
+                    }
+                    if (token.isCancellationRequested) {
+                        this.searchInWorkspaceService.cancel(searchId);
+                    }
+                    if (result && result.matches && result.matches.length) {
+                        this.proxy.$onTextSearchResult(result, searchRequestId);
+                        matches += result.matches.length;
+                        if (options.maxResults && options.maxResults < matches) {
+                            this.searchInWorkspaceService.cancel(searchId);
+                        } else if (defaultMaxHits < matches) {
+                            this.searchInWorkspaceService.cancel(searchId);
+                        }
+                    }
+                },
+                onDone: (searchId, _error) => {
+                    const hasSearch = this.workspaceSearch.has(searchId);
+                    if (hasSearch) {
+                        this.workspaceSearch.delete(searchId);
+                    }
+                    if (options.maxResults && options.maxResults < matches) {
+                        resolve({ limitHit: true });
+                    } else if (defaultMaxHits < matches) {
+                        resolve({ limitHit: true });
+                    }
+                    resolve({ limitHit: false });
```

Comment:
Let's clean this up, and move the `resolve({ limitHit: false });` into an `else`.